certora/specs/Gates.spec

```
@@ -123,27 +68,19 @@ rule cantSendShares(env e, method f, calldataarg args, address user, uint256 sha
 // Check that transfers initiated from the vault, assuming the vault is not reentred, may only increase the balance of a given user when he can't send, and similarly the balance may only decrease when he can't receive.
 // Assume that the vault only uses market V1 and vault V1 adapters, and that those are properly deployed to point to market V1 and vault V1 respectively.
 rule cantSendAssetsAndCantReceiveAssets(env e, method f, calldataarg args, address user) filtered {
-    f -> f.selector != sig:MorphoMarketV1Adapter.skim(address).selector &&
-         f.selector != sig:MorphoVaultV1Adapter.skim(address).selector  &&
-         f.selector != sig:multicall(bytes[]).selector
+    f -> f.selector != sig:multicall(bytes[]).selector
 }{
     // Trick to require that all the following addresses are different.
-    require(MorphoMarketV1Adapter == 0x10, "ack");
-    require(MorphoVaultV1Adapter == 0x11, "ack");
     require(currentContract == 0x12, "ack");
     require(asset() == 0x13, "ack");
 
-    require (user != MorphoMarketV1Adapter &&
-             user != MorphoVaultV1Adapter &&
-             user != MorphoMarketV1Adapter.morpho &&
-             user != MorphoVaultV1Adapter.morphoVaultV1 &&
-             user != currentContract, "do not check if the vault, the adapters themselves, Morpho markets V1 and Morpho vaults V1 are properly gated to not send/receive assets");
-
-    require (forall address adapter. currentContract.isAdapter[adapter] => (adapter == MorphoMarketV1Adapter || adapter == MorphoVaultV1Adapter), "assume that, currently, only MorphoMarketV1Adapter and MorphoVaultV1Adapter are under scrutiny");
+    require (user != currentContract);
+    require (!currentContract.isAdapter[user]);
+    require (!currentContract.isAdapter[asset()]);
```

```suggestion
    require (user != currentContract, "no gate check on called contract");
    require (!currentContract.isAdapter[user], "no gate check on adapters");
    require (!currentContract.isAdapter[asset()], ,"?");
```

Didn't put anything in the last line, I don't know why the third require is necessary.